test/WebSocket.test.js

```diff
@@ -461,6 +461,30 @@ describe('WebSocket', function () {
         req.abort();
       });
     });
+
+    it('error is emitted if server response exceeds timeout', function (done) {
+      const timeout = 100;
+      server.once('upgrade', (req, socket) => {
```

Comment:
Ok, this is becuase `allowHalfOpen` is set to true and the server socket is not automatically closed.

This should fix the issue.

```js
server.once('upgrade', (req, socket) => socket.on('end', socket.end));
```